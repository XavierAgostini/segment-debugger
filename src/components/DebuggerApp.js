import React from 'react'
import DetailsPane from './DetailsPane'
import ListPane from './ListPane'
import { Button, Tabs } from 'evergreen-ui'

const divStyle = {
  display: 'flex'
}

const eventTypeFilter = (events, searchText) => {
  return events.filter(((event) => {
    const eventType = event.type.toLowerCase()
    return searchText.length === 0 || eventType.indexOf(searchText) > -1; 
  }))
}

export default class DebuggerApp extends React.Component {
  constructor(props) {
    super(props)
    this.eventSource = new EventSource('http://localhost:5000/debugger-stream')
  }
  state = {
    isPaused: false,
    selectedEvent: undefined,
    events: [],
    filteredEvents: [],
    searchText: ''
  }
  componentDidMount() {
    console.log('debugger app mounted')
    this.eventSource.addEventListener('message', (event) => {
      if(!this.state.isPaused) {
        this.setState((prevState) => ({
          events: [JSON.parse(event.data)].concat(prevState.events)
        }))
      } 
    })
  }
  
  handlePauseChange = (pauseStatus) => {
    this.setState(() => ({ isPaused: pauseStatus}))
  }

  handleSearchFilter = (searchText) => {
    console.log('handleSearchFilter')
    this.setState(()=>({ searchText }))
  }

  handleEventSelected = (event) => {
    console.log('handleEventSelected')
    this.setState(()=>({ selectedEvent: event }))
  }

  render() {
    return (
      <div style={divStyle}>
        <ListPane
          events={eventTypeFilter(this.state.events, this.state.searchText)}
          handlePauseChange={this.handlePauseChange}
          handleSearchFilter={this.handleSearchFilter}
          handleEventSelected={this.handleEventSelected}
        />
        <DetailsPane selectedEvent={this.state.selectedEvent}/>
      </div>
    )
  }
}