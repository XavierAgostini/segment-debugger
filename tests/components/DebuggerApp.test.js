import React from 'react';
import { shallow } from 'enzyme';
import DebuggerApp from '../../src/components/DebuggerApp'
import testEvents from '../seed/seed'

let wrapper = shallow(<DebuggerApp/>)
describe('<DebuggerApp/>', () => {
  test('should render DebuggerApp as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })
})