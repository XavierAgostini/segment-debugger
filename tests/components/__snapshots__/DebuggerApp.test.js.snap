// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DebuggerApp/> should render DebuggerApp as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DebuggerApp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ListPane
          events={Array []}
          handleEventSelected={[Function]}
          handlePauseChange={[Function]}
          handleSearchFilter={[Function]}
          onClick={[Function]}
        />,
        <DetailsPane />,
      ],
      "style": Object {
        "display": "flex",
        "flex": 1,
        "height": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "events": Array [],
          "handleEventSelected": [Function],
          "handlePauseChange": [Function],
          "handleSearchFilter": [Function],
          "onClick": [Function],
          "selectedEvent": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "selectedEvent": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListPane
            events={Array []}
            handleEventSelected={[Function]}
            handlePauseChange={[Function]}
            handleSearchFilter={[Function]}
            onClick={[Function]}
          />,
          <DetailsPane />,
        ],
        "style": Object {
          "display": "flex",
          "flex": 1,
          "height": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "events": Array [],
            "handleEventSelected": [Function],
            "handlePauseChange": [Function],
            "handleSearchFilter": [Function],
            "onClick": [Function],
            "selectedEvent": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selectedEvent": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
